docs/architecture.md

```diff
@@ -0,0 +1,91 @@
+The events system uses both Dyno types available on Heroku:
+
+- a unique **web** Dyno, exposed as https://events.code-review.moz.tools
+- at least one **worker** Dyno, not reachable publicly
+
+The **web** Dyno only stores HTTP requests from Phabricator into the shared Redis Database (also provided by Heroku as an addon).
+
+Each **worker** clones a list of Mozilla repositories at startup (including the huge [mozilla-central](https://hg.mozilla.org/mozilla-central/)), polls the Redis database for new builds to run.
+
+For each build retrieved, the stack of patches is retrieved from the Phabricator API (and not HGMO), then applied locally on the target repository. Once the stack is cleanly applied, a mercurial push to the corresponding try server is done.
+
+Finally, the Phabricator build is updated with the treeherder link, so that developers can view the progress of their review build while it's running (also useful for debugging !).
+
+The list of repositories, try servers and options to run the Events system on Heroku is stored in [the shared configuration on Taskcluster](configuration.md).
+
+## Analyzers
+
+The whole project relies on the code analyzers defined on the target repositories. They can produce issues detected by static analyzers, or formatting issues, with patches to improve the code quality.
+
+We currently support several analyzers used in Mozilla central:
+
+- clang-format
+- clang-tidy
+- coverity
+- infer
+- mozlint (bringing a huge list of analyzers like flake8, eslint, rustfmt, ...)
+
+All those analyzers output a different JSON payload, hence the need for different implementations. But we also have a [default format](analysis_format.md) that is recommended if you want to add a new analyzer.
+
+In an ideal world, we would only support that format, and convert the existing analyzers to output a compatible payload.
+
+## Publication task
+
+You may have noticed the *pulse message dance* in the workflow above. It may seem weird to add a task only sending a Pulse message to trigger another task in the same Taskcluster instance.
+
+This has brought us several advantages:
+- we do not [ever](https://hg.mozilla.org/mozilla-central/log/tip/taskcluster/ci/code-review/kind.yml) need to update the first task, so we do not have a direct dependency on mozilla-central
+- we can update the publication code (the **bot** part) easily, just like any other Taskcluster hook payload
+- we can trigger the hook from different projects, with different rules and workflows
+- we get a trigger history for free thanks to Taskcluster hook system
+- we can run several versions of the hook in parallel (testing environment can receive the same pulse messages than production, allowing to run in *shadow mode* - see [debugging](debugging.md) for that)
+
+## Backend
+
+The backend is hosted on a single Heroku web dyno as https://api.code-review.moz.tools and linked to an Heroku postgres database. This setup would allow us to scale easily if the need arise.
```

Comment:
```suggestion
The backend is hosted on a single Heroku web dyno at https://api.code-review.moz.tools and linked to an Heroku Postgres database. This setup would allow us to scale easily if the need arise.
```